server/src/main/java/org/apache/iotdb/db/qp/strategy/LogicalGenerator.java

```diff
@@ -1475,7 +1475,15 @@ long parseTimeFormat(String timestampStr) throws SQLParserException {
       throw new SQLParserException("input timestamp cannot be empty");
     }
     if (timestampStr.equalsIgnoreCase(SQLConstant.NOW_FUNC)) {
-      return System.currentTimeMillis();
+      String timePrecision = IoTDBDescriptor.getInstance().getConfig().getTimestampPrecision();
+      switch (timePrecision) {
+        case "ns":
+          return System.currentTimeMillis() * 1000_000L;
```

Comment:
I'm just a little bit curious why don't you use `System.nanoTime()` here.

Reply:
When users use `now()` and set the precision as `ns` or `ms`, they want to generate dense data. if you you `System.currentTimeMillis()*1000` or `1000_000`, then many timstamps will repeat...

Reply:
System.nanoTime is not the current time in nano second, it is usually used for timing:

long start = System.nanoTime();
long elapse = System.nanoTime() - start;

And, there is no System. microsecond()...

Reply:
I agree with @qiaojialin and @jixuan1989 both.  System.nanoTime can't get the current time correctly, while the precision of *1000 or *1000_100 is unchanged actually. We need to rethink about the implementation way.

Reply:
Use Instant to generate nano seconds instead.

Reply:
I think `Instant.now()` is a good solution. 
Another way is just using the last 6 digits of System.nanoTime(), as in a JVM instance (which means if you do not restart the JVM), System.nanoTime() is keeping increasing.

Reply:
`Instant.getNano()` returns the nanoseconds, but actully it's still millisecond resolution. I'm not sure it's useful here.

Reply:
I think what we need is to keep the nano time as monotonic increasing.